.circleci/config.yml

```diff
@@ -84,7 +86,361 @@ commands:
           name: "Wait for IPFS daemon to start"
           command: wget --retry-connrefused --waitretry=20 --read-timeout=20 --timeout=15 -t 10 --post-data '' "http://localhost:5001/api/v0/version"
 
+  version-contracts:
+    steps:
+      # Find the last update of any contract
+      - run: |
+          git log -1 -- \
+            utils/core-modules/contracts \
+            utils/core-contracts/contracts \
+            protocol/synthetix/contracts \
+            protocol/governance/contracts \
+            protocol/oracle-manager/contracts \
+            markets/spot-market/contracts \
+            markets/perps-market/contracts \
+            markets/legacy-market/contracts \
+            > /tmp/version--contracts.txt
+      - run: cat /tmp/version--contracts.txt
+
+  version-testable-contracts:
+    steps:
+      # Currently we only have testable contracts generated for @synthetixio/main
+      # Add more folders to version calc (space separated)
+      - run: git log -1 -- protocol/synthetix/contracts > /tmp/version--testable-contracts.txt
+      - run: cat /tmp/version--testable-contracts.txt
+
 jobs:
+  build-testable:
+    docker:
+      - image: cimg/node:<< pipeline.parameters.node-version >>
+    steps:
+      - checkout
+      - install-foundry
+      - yarn-install
+      - run: yarn workspaces foreach --topological-dev --verbose run build:ts
+
+      - restore_cache:
+          keys:
+            - hardhat-{{ .Environment.SOLC_VERSION }}
+
+      - version-contracts
+      - restore_cache:
+          keys:
+            - cannon-{{ .Environment.CACHE_VERSION }}-{{ checksum "/tmp/version--contracts.txt" }}
+
```

Comment:
Cannon builds can be greatly increased in speed if we load an inexact cache , since only changes contracts will be rebuilt into the image. Would reccomend loading from additional cache prefixes even if contracts chance

Reply:
Aka adding a second restore key with the checksum removed should do the trick